shell/platform/embedder/embedder.cc

```diff
@@ -815,6 +822,47 @@ static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
 #endif
 }
 
+static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
+    GrDirectContext* context,
+    const FlutterBackingStoreConfig& config,
+    const FlutterOpenGLSurface* surface) {
+#ifdef SHELL_ENABLE_GL
+  GrGLFramebufferInfo framebuffer_info = {};
+  framebuffer_info.fFormat = GL_BGRA8_EXT;
```

Comment:
Can't the surface format come from the config too? Why assume something here?

Reply:
True, but then the kN32_SkColorType below must be chosen correctly as well, otherwise `WrapBackendRenderTarget` fails.

But what does that format info even affect precisely? In my experience it didn't really do anything, I've just tested `GL_RGBA8` and `GL_BGRA8_EXT` with an EGL surface that uses `DRM_FORMAT_ARGB8888` and it both works, even `DRM_FORMAT_RGB565` surfaces just work with whatever GL format you give to skia. Is that format specifed there only used for pixel readback or something?

Reply:
> True, but then the kN32_SkColorType below must be chosen correctly as well, otherwise WrapBackendRenderTarget fails.


Thats fair. We can add in the format at a later point if needed.

Reply:
> Thats fair. We can add in the format at a later point if needed.

:+1: I think that also makes sense with impeller coming up soon, the list of supported pixel formats might change a bit then as well.